pennylane_lightning/core/src/simulators/lightning_tensor/tncuda/measurements/MeasurementsTNCuda.hpp

```diff
@@ -164,6 +165,97 @@ template <class TensorNetT> class MeasurementsTNCuda {
+            /* int32_t numModesToSample */ modesToSample.size(),
+            /* const int32_t *modesToSample */ modesToSample.data(),
+            /* cutensornetStateSampler_t * */ &sampler));
+
+        // Configure the quantum circuit sampler
+        const cutensornetSamplerAttributes_t samplerAttributes =
+            CUTENSORNET_SAMPLER_CONFIG_NUM_HYPER_SAMPLES;
+
+        PL_CUTENSORNET_IS_SUCCESS(cutensornetSamplerConfigure(
+            /* const cutensornetHandle_t */ tensor_network_.getTNCudaHandle(),
+            /* cutensornetStateSampler_t */ sampler,
+            /* cutensornetSamplerAttributes_t */ samplerAttributes,
+            /* const void *attributeValue */ &numHyperSamples,
+            /* size_t attributeSize */ sizeof(numHyperSamples)));
+
+        cutensornetWorkspaceDescriptor_t workDesc;
+        PL_CUTENSORNET_IS_SUCCESS(cutensornetCreateWorkspaceDescriptor(
+            /* const cutensornetHandle_t */ tensor_network_.getTNCudaHandle(),
+            /* cutensornetWorkspaceDescriptor_t * */ &workDesc));
+
+        const std::size_t scratchSize = cuUtil::getFreeMemorySize() / 2;
+
+        // Prepare the quantum circuit sampler for sampling
+        PL_CUTENSORNET_IS_SUCCESS(cutensornetSamplerPrepare(
+            /* const cutensornetHandle_t */ tensor_network_.getTNCudaHandle(),
+            /* cutensornetStateSampler_t */ sampler,
+            /* size_t maxWorkspaceSizeDevice */ scratchSize,
+            /* cutensornetWorkspaceDescriptor_t */ workDesc,
+            /* cudaStream_t unused as of v24.08 */ 0x0));
+
+        std::size_t worksize =
+            getWorkSpaceMemorySize(tensor_network_.getTNCudaHandle(), workDesc);
+
+        PL_ABORT_IF(worksize > scratchSize,
+                    "Insufficient workspace size on Device.\n");
+
+        const std::size_t d_scratch_length = worksize / sizeof(size_t) + 1;
+        DataBuffer<std::size_t, int> d_scratch(
+            d_scratch_length, tensor_network_.getDevTag(), true);
+
+        setWorkSpaceMemory(tensor_network_.getTNCudaHandle(), workDesc,
+                           reinterpret_cast<void *>(d_scratch.getData()),
+                           worksize);
+
+        PL_CUTENSORNET_IS_SUCCESS(cutensornetSamplerSample(
+            /* const cutensornetHandle_t */ tensor_network_.getTNCudaHandle(),
+            /* cutensornetStateSampler_t */ sampler,
+            /* int64_t numShots */ num_samples,
+            /* cutensornetWorkspaceDescriptor_t */ workDesc,
+            /* int64_t * */ samples.data(),
+            /* cudaStream_t unused as of v24.08  */ 0x0));
+
+        PL_CUTENSORNET_IS_SUCCESS(
+            cutensornetDestroyWorkspaceDescriptor(workDesc));
+        PL_CUTENSORNET_IS_SUCCESS(cutensornetDestroySampler(sampler));
+
+        std::vector<std::size_t> samples_size_t(samples.size());
+
+        std::transform(samples.begin(), samples.end(), samples_size_t.begin(),
+                       [&](int64_t x) { return static_cast<std::size_t>(x); });
```

Comment:
I don't think we need to pass any local variables by reference here. What do you think?
```suggestion
                       [](int64_t x) { return static_cast<std::size_t>(x); });
```

Reply:
Nice spot!